cancer_ai/validator/utils.py

```diff
@@ -173,60 +173,93 @@ async def fetch_yaml_data_from_local_repo(local_repo_path: str) -> list[dict]:
+    """
+    all_orgs = []
     for file in fetched_yaml_files:
-        yaml_data = file['yaml_data']
-        date_uploaded = file['date_uploaded']
-
-        org_id = yaml_data[0]['organization_id']
-        existing_org = next((org for org in factory.organizations if org.organization_id == org_id), None)
-
-        if not existing_org:
-            for entry in yaml_data:
-                data_references.append(DatasetReference(
-                    competition_id=entry['competition_id'],
-                    dataset_hf_repo=entry['dataset_hf_repo'],
-                    dataset_hf_filename=entry['dataset_hf_filename'],
-                    dataset_hf_repo_type=entry['dataset_hf_repo_type'],
-                    dataset_size=entry['dataset_size']
-                ))
-        
-        if existing_org and date_uploaded != existing_org.date_uploaded:
-            last_entry = yaml_data[-1]
-            data_references.append(DatasetReference(
-                competition_id=last_entry['competition_id'],
-                dataset_hf_repo=last_entry['dataset_hf_repo'],
-                dataset_hf_filename=last_entry['dataset_hf_filename'],
-                dataset_hf_repo_type=last_entry['dataset_hf_repo_type'],
-                dataset_size=last_entry['dataset_size']
-            ))
-
-    return data_references
-
-async def update_organizations_data_references(fetched_yaml_files: list[dict]):
-    bt.logging.trace("Updating organizations data references")
+        yaml_data = file["yaml_data"]  # each yaml_data is a list of dicts
+        for entry in yaml_data:
+            # Remap 'org_id' to 'organization_id' if needed.
+            if "org_id" in entry:
+                entry["organization_id"] = entry.pop("org_id")
+            all_orgs.append(entry)
+    
+    # Prepare the dictionary to update the factory.
+    update_data = {"organizations": all_orgs}
+    
+    # Update the singleton instance.
     factory = OrganizationDataReferenceFactory.get_instance()
-    factory.organizations.clear()
+    factory.update_from_dict(update_data)
 
-    for file in fetched_yaml_files:
-        yaml_data = file['yaml_data']
-        new_org = OrganizationDataReference(
-            organization_id=yaml_data[0]['organization_id'],
-            contact_email=yaml_data[0]['contact_email'],
-            bittensor_hotkey=yaml_data[0]['bittensor_hotkey'],
-            data_packages=[
-                DatasetReference(
-                    competition_id=dp['competition_id'],
-                    dataset_hf_repo=dp['dataset_hf_repo'],
-                    dataset_hf_filename=dp['dataset_hf_filename'],
-                    dataset_hf_repo_type=dp['dataset_hf_repo_type'],
-                    dataset_size=dp['dataset_size']
-                )
-                for dp in yaml_data
-            ],
-            date_uploaded=file['date_uploaded']
+async def check_for_new_dataset_files(hf_api: HfApi, org_latest_updates: dict) -> list[NewDatasetFile]:
+    """
```

Comment:
I don't understand this code, can you refactor a bit or add revelant comments what is happening where?